src/shared/gcpagent/agent.go

```diff
@@ -0,0 +1,63 @@
+package gcpagent
+
+import (
+	"cloud.google.com/go/compute/metadata"
+	"context"
+	"github.com/otterize/intents-operator/src/shared/errors"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/viper"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	EnvGcpProjectId = "gcp-project-id"
+	EnvGcpGkeName   = "gcp-project-id"
+)
+
+type Agent struct {
+	projectID   string
+	clusterName string
+	client      client.Client
+}
+
+func NewGCPAgent(ctx context.Context, c client.Client) (*Agent, error) {
+	logrus.Info("Initializing GCP Intents agent")
+
+	// Get the current GCP project using the metadata server or local env
+	projectID, err := getGCPAttribute(EnvGcpProjectId)
+	if err != nil {
+		return nil, errors.Errorf("failed to get current GCP project: %w", err)
+	}
+
+	// Retrieve the cluster name using the metadata server or local env
+	clusterName, err := getGCPAttribute(EnvGcpGkeName)
+	if err != nil {
+		return nil, errors.Errorf("failed to get current GKE cluster: %w", err)
+	}
+
+	return &Agent{
+		client:      c,
+		projectID:   projectID,
+		clusterName: clusterName,
+	}, nil
+}
+
+func getGCPAttribute(attribute string) (res string, err error) {
+	if attribute == EnvGcpProjectId {
+		res, err = metadata.ProjectID()
+		if err == nil {
+			return res, nil
+		}
+	} else if attribute == EnvGcpGkeName {
+		res, err = metadata.InstanceAttributeValue("cluster-name")
+		if err == nil {
+			return res, nil
+		}
+	}
```

Comment:
consider using `switch` (it'll be more readable and more idiomatic in go in this case)

Reply:
done